src/compiler/emitter.ts

```diff
@@ -973,6 +1019,11 @@ namespace ts {
             }
         }
 
+        // SyntaxKind.UnparsedSource
+        function emitUnparsedSource(unparsed: UnparsedSource) {
+            write(unparsed.javascriptText);

```

Comment:
Probably also want a `writeLine`, unless we know all prepends are guaranteed to have a trailing `newLine`?

Reply:
Also: This emitter is used by both the declarations and js - something needs to swap this to using the `declarationText` member (instead of javascriptText) during declaration emit (as i doubt you want your JS concatenated into your declarations). What I'd consider doing is feeding the `prepends` into the transforms (ts and declaration - looks like they already are kinda, since they're part of the bundle) and having the transformer rewrite the prepend data as the appropriate `UnparsedSource` node. What I mean by this, is that we have a `UnparsedSourceData` node that has all of js, declaration, and sourcemap(s) info (what `UnparsedSource` is now), which is given as part of the bundle into the transformer pipeline, and then the `ts` transform converts it into an `UnparsedSource` node (which simply has a generic `text` and `sourceMaps` property) containing just the js info, and the `declarations` transformer does the same, but for the declaration text and maps.